### ISSUE TITLE
Qualitative Analysis v2

### ISSUE THREAD

**Issue by pavlovcik** [ubiquibot/comment-incentives#9]
- Cloudflare offers a free vector embeddings worker template which uses `cf/baai/bge-base-en-v1.5` [^1^]
- You can access it here https://dash.cloudflare.com/WORKSPACE_ID/workers-and-pages/create/workers/new?template=vector-embedding-app
   - Change `WORKSPACE_ID` to your Cloudflare workspace id. 

I've never worked with vector embeddings but I understand the general concept of it. We should use Cloudflare's free service to generate and store embeddings of every GitHub comment when the comment incentives are being calculated.

I think that we can make a `public.vector-embeddings` table, and the ID can be the GitHub comment ID. Then we can store an embedding in the next column over. 

Finally I understand that we should be able to compute how "on topic" a comment is, based on the GItHub issue specification that the comment was posted on. 

### R&D Tasks
0. Determine if Supabase's solution is significantly superior. If not, I would default to Cloudflare's offering because they have amazing infrastructure. 
1. Perform a manual prototype to confirm that this approach works moderately well for computing exactly how on topic a comment is. Minimal glue code should be produced and the data should be manipulated mostly manually at this stage.
2. Set up Cloudflare Worker with `cf/baai/bge-base-en-v1.5`
3. Set up Supabase `public.vector-embeddings` table with `id` and `embedding` 
4. Deploy prototype.
5. Connect prototype to our comment incentives calculations. 

[^1^]: https://huggingface.co/BAAI/bge-base-en-v1.5

**Comment by pavlovcik** [comment]
@FernandVEYRIER if you want to work on this repository, perhaps instead you can work on this task but package it as a standalone module. Then we can import it in the [new version](https://github.com/ubiquibot/conversation-rewards) once we get that far with it?

**Comment by FernandVEYRIER** [comment]
@pavlovcik It feels like this repo should eventually be deprecated in favor of the newer version isn't it?

**Comment by pavlovcik** [comment]
Yes but the research will carry over. Also the deliverables can be packaged neatly into portable modules that we can use in the next version!

**Comment by FernandVEYRIER** [comment]
Sure thing. This should be broken down into smaller tasks to be carried on then

**Comment by pavlovcik** [comment]
It is. Start with #11 and then we will generate the rest based on the research results of #11
